pyproject.toml

```diff
@@ -1,3 +1,107 @@
+[project]
+name = "charmcraft"
+description = "The main tool to build, upload, and develop in general the Juju charms."
+dynamic = ["version", "readme"]
+dependencies = [
+    "craft-application~=1.1",
+    "craft-cli>=2.3.0",
+    "craft-parts>=1.18",
+    "craft-providers",
+    "craft-store>=2.4",
+    "distro>=1.3.0",
+    "humanize>=2.6.0",
+    "jsonschema",
+    "jinja2",
+    # Pydantic will need to be updated all at once with our libraries.
+    # When you update it, remove the pydantic constraints from renovate.
+    "pydantic>=1.10,<2.0",
+    "python-dateutil",
+    "pyyaml",
+    "requests",
+    "requests-toolbelt",
+    "requests-unixsocket",
+    "snap-helpers",
+    "tabulate",
+    # Needed until requests-unixsocket supports urllib3 v2
+    # https://github.com/msabramo/requests-unixsocket/pull/69
+    "urllib3<2.0",
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Environment :: Console",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: Apache Software License",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+]
+requires-python = ">=3.10"
+
+[project.scripts]
+
+[project.optional-dependencies]
+dev = [
+    "coverage",
+    "flake8",
```

Comment:
Should we move this to lint?

Reply:
Thanks for bringing this up! We don't use flake8 or pydocstyle for linting charmcraft, but they're currently still used for testing the templates from `charmcraft init`, so this is the right place for them.

Given that the templates are [moving to using ruff](https://github.com/canonical/charmcraft/pull/1378), we should replace those tests with a test that runs ruff on them instead.

I've created an enhancement request for that: https://github.com/canonical/charmcraft/issues/1461